spring-xd-dirt/src/main/java/org/springframework/xd/dirt/module/store/ZooKeeperComposedModuleDefinitionRegistry.java

```diff
@@ -163,9 +176,12 @@ public ModuleDefinition findDefinition(String name, ModuleType type) {
 						Paths.build(Paths.MODULES, type.toString(), child));
 				// Check for data (only composed modules have definitions)
 				if (data != null && data.length > 0) {
-
 					ModuleDefinition composed = this.findDefinition(child, type);
-					results.add(composed);
+					if (composed != null) {
+						results.add(composed);
+					} else {
```

Comment:
Coding format doesn't conform to Spring standard: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Code-Style. Will fix while merging.